src/_pytest/doctest.py

```diff
@@ -536,6 +537,21 @@ def _find(
                         tests, obj, name, module, source_lines, globs, seen
                     )
 
+        class CachedPropertyAwareDocTestFinder(MockAwareDocTestFinder):
+            def _from_module(self, module, object):
+                """Doctest code does not take into account `@cached_property`,
+                this is a hackish way to fix it. https://github.com/python/cpython/issues/107995
+
+                Wrap Doctest finder so that when it calls `_from_module` for
+                a cached_property it uses the underlying function instead of the
+                wrapped cached_property object.
+                """
+                if isinstance(object, functools.cached_property):
+                    object = object.func
+
+                # Type ignored because this is a private function.
+                return super()._from_module(module, object)  # type: ignore[misc]
+
         if self.path.name == "conftest.py":
```

Comment:
At first glance this Looks like this ought to be a extra elif next two the property handling of the mock class with the hack's

Reply:
Yeah that makes perfect sense. I'm going to wait on the cpython fix to be implemented before pushing this forward. It might offer some additional insights.

Reply:
https://github.com/python/cpython/pull/107996 has been merged. I think the original idea of the PR is still the way to fix this issue. However, I did move the method override of `_from_module` to the pre-existing `DocTestFinder` subclass.

Reply:
Should we rename [`MockAwareDocTestFinder`](https://github.com/tjsmart/pytest/blob/ebd571bb18d2cf00ebc70db8090217fc62d00e98/src/_pytest/doctest.py#L502)?